bitmovin-analytics-datasource/src/components/FilterRow.tsx

```diff
@@ -1,136 +1,135 @@
-    setFilters((prevState) => [
-      ...prevState,
-      {
-        selectedAttribute: {},
-        selectedOperator: {},
-        rawFilterValue: '',
-        convertedFilterValue: '',
-        parsingValueError: '',
-      } as Filter,
-    ]);
+    const newFilters = [...props.filters];
+    newFilters.push({
+      attribute: {},
+      operator: {},
+      rawFilterValue: '',
+      convertedFilterValue: '',
+      parsingValueError: '',
+    } as FilterRowData);
+    props.onFilterRowChange(newFilters);
   };
 
   const onAddFilter = (index: number) => {
-    const filter = filters[index];
+    const filter = props.filters[index];
     try {
       const convertedValue = convertFilterValueToProperType(
         filter.rawFilterValue,
-        filter.selectedAttribute.value!,
-        filter.selectedAttribute.label!,
-        filter.selectedOperator.value!,
+        filter.attribute,
+        filter.attribute,
+        filter.operator,
         props.isAdAnalytics
       );
 
-      const newFilter = { ...filter, convertedFilterValue: convertedValue, parsingValueError: '' } as Filter;
+      const newFilter = { ...filter, convertedFilterValue: convertedValue, parsingValueError: '' } as FilterRowData;
 
-      const newFilters = [...filters];
+      const newFilters = [...props.filters];
       newFilters.splice(index, 1, newFilter);
 
-      setFilters(newFilters);
+      props.onFilterRowChange(newFilters);
 
-      props.onChange(mapFiltersToQueryFilters(newFilters));
+      props.onQueryFilterChange(mapFilterRowsToQueryFilters(newFilters));
     } catch (e: unknown) {
       if (e instanceof Error) {
         const errorMessage = e.message;
-        const newFilter = { ...filter, parsingValueError: errorMessage } as Filter;
+        const newFilter = { ...filter, parsingValueError: errorMessage } as FilterRowData;
 
-        const newFilters = [...filters];
+        const newFilters = [...props.filters];
         newFilters.splice(index, 1, newFilter);
 
-        setFilters(newFilters);
+        props.onFilterRowChange(newFilters);
       }
     }
   };
 
   const deleteFilterInput = (index: number) => {
-    const newFilters = [...filters];
+    const newFilters = [...props.filters];
     newFilters.splice(index, 1);
 
-    setFilters(newFilters);
+    props.onFilterRowChange(newFilters);
 
-    props.onChange(mapFiltersToQueryFilters(newFilters));
+    props.onQueryFilterChange(mapFilterRowsToQueryFilters(newFilters));
   };
 
   const onAttributesChange = (index: number, newAttribute: SelectableValue<QueryAttribute | QueryAdAttribute>) => {
-    const filter = filters[index];
-    const newFilter = { ...filter, selectedAttribute: newAttribute } as Filter;
-    const newFilters = [...filters];
+    const filter = props.filters[index];
+    const newFilter = { ...filter, attribute: newAttribute.value } as FilterRowData;
+    const newFilters = [...props.filters];
     newFilters.splice(index, 1, newFilter);
```

Comment:
maybe lodash has something more human readable instead of `.splice(index, 1, newFilter);`